### ISSUE TITLE
Constant traits not working for integer types

### ISSUE THREAD

**Issue by Speierers** [mitsuba-renderer/drjit#226]
The constant traits in `const.py` and `array_constants.h` do not handle integer types properly as they assume floating-point types.

For the traits specific to floats (e.g. `dr::Pi`, `dr::Epsilon`) we should probably add a static assert, and other traits (e.g. `dr::Smallest`, `dr::Largest`, ...) should be fixed.

If I find the time, I will try to open a PR to fix those issues.

**Comment by merlinND** [comment]
Hi @Speierers,

Just in case you're referring to the non-nanobind version, before opening a PR, check out the upcoming version, which I believe is implemented differently (don't know how it handles integer types though): https://github.com/mitsuba-renderer/drjit/blob/nanobind_v2/include/drjit/array_constants.h

**Comment by Speierers** [comment]
Good point @merlinND 👍  although I think the issue is still there in the nanobind version.
